app/src/test/java/br/com/catbag/gifreduxsample/reducers/reducers/GifListReducerTest.java

```diff
@@ -37,85 +39,82 @@
 import static junit.framework.Assert.assertTrue;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
-import static shared.TestUtils.DEFAULT_UUID;
-import static shared.TestUtils.buildGif;
-import static shared.TestUtils.createDefaultImmutableGif;
-import static shared.TestUtils.createStateFromGif;
-import static shared.TestUtils.createStateFromGifs;
-import static shared.TestUtils.getFakeGifs;
-import static shared.TestUtils.getFirstGif;
-import static shared.TestUtils.getFluxxan;
+import static shared.TestHelper.DEFAULT_UUID;
+import static shared.TestHelper.buildAppState;
+import static shared.TestHelper.buildFiveGifs;
+import static shared.TestHelper.buildGif;
+import static shared.TestHelper.builderWithDefault;
 
 // Roboeletric still not supports API 24 stuffs
 @Config(sdk = 23, constants=BuildConfig.class)
 @RunWith(RobolectricTestRunner.class)
 public class GifListReducerTest {
-    private Boolean mStateChanged = false;
+    private TestHelper helper
+            = new TestHelper(((MyApp) RuntimeEnvironment.application).getFluxxan());
+
+    @Rule
+    public Timeout mGlobalTimeout = new Timeout(20, TimeUnit.SECONDS);
 
     @Before
-    public void setup(){
-        getFluxxan().registerReducer(new FakeReducer());
-        getFluxxan().getDispatcher().addListener(new StateListener() {
-            @Override
-            public boolean hasStateChanged(Object newState, Object oldState) {
-                mStateChanged = true;
-                return newState != oldState;
-            }
-
-            @Override
-            public void onStateChanged(Object o) {
-            }
-        });
+    public void setup() {
+        helper.getFluxxan().registerReducer(new FakeReducer());
+        helper.activateStateListener();
     }
 
-    @Test(timeout = 1000)
+    @After
+    public void cleanup() {
+        helper.getFluxxan().registerReducer(new FakeReducer());
+        helper.deactivateStateListener();
+    }
```

Comment:
This before and after methods are duplcated between tests, maybe we can move them to some base test class.